src/test/article.spec.js

```diff
@@ -0,0 +1,107 @@
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+import 'chai/register-should';
+import 'chai/register-expect';
+import sinon from 'sinon';
+import rewire from 'rewire';
+import slug from 'slug';
+
+import app from '../index';
+import models from '../database/models';
+import { articleSample } from '../database/mockdata/articledata';
+
+const { User, Article } = models;
+
+// configure chai to use expect
+chai.use(chaiHttp);
+const { expect } = chai;
+
+const articleEndpoint = '/api/v1/articles';
+const specificArticleEndpoint = '/api/v1/articles/:articleId';
+
+describe('Articles', () => {
+  describe('CREATE ARTICLE', () => {
+    before(() => {});
+
+    it('should create article successfully, with valid user input', done => {
+      chai
+        .request(app)
+        .post(articleEndpoint)
+        .send(articleSample)
+        .end((error, response) => {
+          expect(response.status).to.equal(201);
+          done();
+        });
+    });
+  });
+
+  describe('GET ALL ARTICLES', () => {
+    before(() => {});
+
+    it('should fetch all successfully created articles', done => {
```

Comment:
Hello @tobechukwuobitube , could you check tests already on the repository for examples, these tests look dependent on one another hence disobeying the rule of testing. For example, where is the article you are getting on line 130 coming from, what happens if I remove all the test before it, would this test still return the same result?

Reply:
Thank you for this review.
For the reference you made on `line 130`, this test is set up to ensure that the number of articles returned by the application is equal to the number of articles in the database at any instance.
This improves the confidence of the application since it is testing the behaviour of the application.